dashboard/app/jobs.go

```diff
@@ -567,42 +569,41 @@ func createJobResp(c context.Context, job *Job, jobKey *db.Key) (*dashapi.JobPol
 // miss the situation when some actual user sends a patch testing request without
 // patch.
 func isRetestReproJob(job *Job, build *Build) bool {
-	return job.Type == JobTestPatch &&
+	return (job.Type == JobTestPatch || job.Type == JobBisectFix) &&
 		job.Patch == 0 &&
 		job.KernelRepo == build.KernelRepo &&
 		job.KernelBranch == build.KernelBranch
 }
 
-func handleRetestedRepro(c context.Context, now time.Time, job *Job, jobKey *db.Key, req *dashapi.JobDoneReq) error {
+func handleRetestedRepro(c context.Context, now time.Time, job *Job, jobKey *db.Key,
+	lastBuild *Build, req *dashapi.JobDoneReq) error {
 	bugKey := jobKey.Parent()
 	crashKey := db.NewKey(c, "Crash", "", job.CrashID, bugKey)
 	crash := new(Crash)
 	if err := db.Get(c, crashKey, crash); err != nil {
 		return fmt.Errorf("failed to get crash: %v", crashKey)
 	}
-	build, err := loadBuild(c, job.Namespace, crash.BuildID)
-	if err != nil {
-		return fmt.Errorf("failed to query build: %w", err)
-	}
-	if !isRetestReproJob(job, build) {
-		return nil
-	}
 	bug := new(Bug)
 	if err := db.Get(c, bugKey, bug); err != nil {
 		return fmt.Errorf("failed to get bug: %v", bugKey)
 	}
 	if !config.Namespaces[bug.Namespace].RetestRepros {
 		return nil
 	}
+	allTitles := gatherCrashTitles(req)
 	// Update the crash.
 	crash.LastReproRetest = now
-	allTitles := gatherCrashTitles(req)
-	if req.Error == nil && len(allTitles) == 0 {
-		// If there was any crash at all, the repro is still not worth discarding.
-		// Also, if repro testing itself failed, it might be just a temporary issue.
-		crash.ReproIsRevoked = true
+	if req.Error == nil {
+		// If repro testing itself failed, it might be just a temporary issue.
+		if job.Type == JobTestPatch {
+			// If there was any crash at all, the repro is still not worth discarding.
+			crash.ReproIsRevoked = len(allTitles) == 0
```

Comment:
I think we can flip ReproIsRevoked from true to false here since we do this for retest requests, patch testing requests and fix bisections. Do we want this? Can it break anything? Maybe we check `if !crash.ReproIsRevoked` somewhere in the beginning? If it's already revoked there is nothing to do, right?

Reply:
Hmm, at least at the moment I don't see how the resetting of `ReproIsRevoked` can break things. Though it indeed doesn't seem necessary and it's just safer to deny it. Updated the code.